images/macos/provision/core/openjdk.sh

```diff
@@ -1,53 +1,94 @@
+    archivePath=$(echo ${asset} | jq -r '.binary.package.link')
+    fullVersion=$(echo ${asset} | jq -r '.version.semver' | tr '+' '-')
 
-    javaToolcacheVersionPath=$JAVA_TOOLCACHE_PATH/$fullVersion
-    javaToolcacheVersionArchPath=$javaToolcacheVersionPath/x64
+    JAVA_TOOLCACHE_PATH=${AGENT_TOOLSDIRECTORY}/Java_${VENDOR_NAME}_jdk
+
+    javaToolcacheVersionPath=$JAVA_TOOLCACHE_PATH/${fullVersion}
+    javaToolcacheVersionArchPath=${javaToolcacheVersionPath}/x64
 
     # Download and extract Java binaries
-    download_with_retries $archivePath /tmp OpenJDK${JAVA_VERSION}.tar.gz
-    mkdir -p $javaToolcacheVersionArchPath
-    tar -xzf /tmp/OpenJDK${JAVA_VERSION}.tar.gz -C $javaToolcacheVersionArchPath --strip-components=1
-    # Create complete file
-    touch $javaToolcacheVersionPath/x64.complete
+    download_with_retries ${archivePath} /tmp OpenJDK-${VENDOR_NAME}-${fullVersion}.tar.gz
+    
+    echo "Creating ${javaToolcacheVersionArchPath} directory"
+    mkdir -p ${javaToolcacheVersionArchPath}
 
-    createEnvironmentVariable $JAVA_VERSION $javaToolcacheVersionArchPath
+    tar -xf /tmp/OpenJDK-${VENDOR_NAME}-${fullVersion}.tar.gz -C ${javaToolcacheVersionArchPath} --strip-components=1
+    # Create complete file
+    touch ${javaToolcacheVersionPath}/x64.complete
 
     # Create a symlink to '/Library/Java/JavaVirtualMachines'
     # so '/usr/libexec/java_home' will be able to find Java
-    sudo ln -sf $javaToolcacheVersionArchPath /Library/Java/JavaVirtualMachines/adoptopenjdk-${JAVA_VERSION}.jdk
+    sudo ln -sf ${javaToolcacheVersionArchPath} /Library/Java/JavaVirtualMachines/${VENDOR_NAME}-${JAVA_VERSION}.jdk
 }
 
-JAVA_VERSIONS_LIST=($(get_toolset_value '.java.versions | .[]'))
-JAVA_DEFAULT=$(get_toolset_value '.java.default')
+defaultVersion=$(get_toolset_value '.java.default')
+defaultVendor=$(get_toolset_value '.java.default_vendor')
+jdkVendors=$(get_toolset_value '.java.vendors[].name')
+jdkVersionsToInstall=($(get_toolset_value '.java.vendors[].versions[]'))
 
-for JAVA_VERSION in "${JAVA_VERSIONS_LIST[@]}"
-do
-    installJavaFromAdoptOpenJDK $JAVA_VERSION 
+for jdkVendor in ${jdkVendors}; do
+
+     # remove possible version duplicates
+     jdkVersionsToInstall=$(echo $jdkVersionsToInstall | printf "%s\n" "${jdkVersionsToInstall[@]}" | sort -u)
+     isDefaultVendor=False; [[ ${jdkVendor} == ${defaultVendor} ]] && isDefaultVendor=True
+
+     for jdkVersionToInstall in ${jdkVersionsToInstall}; do
+	    isDefaultVersion=False; [[ ${jdkVersionToInstall} == ${defaultVersion} ]] && isDefaultVersion=True
+
+	    installOpenJDK ${jdkVersionToInstall} ${jdkVendor}
+
+	    if [[ ${isDefaultVendor} == "True" && ${isDefaultVersion} == "True" ]]; then
+		    createEnvironmentVariable ${jdkVersionToInstall} ${jdkVendor} "True"
+	    elif [[ ${isDefaultVendor} == "True" && ${isDefaultVersion} == "False" ]]; then
+		    createEnvironmentVariable ${jdkVersionToInstall} ${jdkVendor}
+	    fi
```

Comment:
I think we should:
- remove line 67
- swap 70 and 72 line
- cut the condition
```
if [[ ${jdkVendor} == ${defaultVendor}]]; then
     createEnvironmentVariable ${jdkVersionToInstall} ${jdkVendor} ${isDefaultVersion}
fi
```